File: sample-java/src/main/java/social/bigbone/sample/StreamFederatedPublicTimeline.java
package social.bigbone.sample;

import social.bigbone.MastodonClient;
import social.bigbone.api.exception.BigBoneRequestException;

import java.io.Closeable;
import java.io.IOException;

public class StreamFederatedPublicTimeline {
    public static void main(final String[] args) throws BigBoneRequestException, InterruptedException, IOException {
        final String instance = args[0];
        final String accessToken = args[1];

        // Instantiate client
        final MastodonClient client = new MastodonClient.Builder(instance)
                .accessToken(accessToken)
                .build();

        // Start federated timeline streaming and stop after 20 seconds
        try (Closeable ignored = client.streaming().federatedPublic(false, System.out::println)) {
            Thread.sleep(20_000L);
        }
    }
}

File: sample-kotlin/src/main/kotlin/social/bigbone/sample/StreamFederatedPublicTimeline.kt
package social.bigbone.sample

import social.bigbone.MastodonClient

object StreamFederatedPublicTimeline {
    @JvmStatic
    fun main(args: Array<String>) {
        val instance = args[0]
        val accessToken = args[1]

        // Instantiate client
        val client = MastodonClient.Builder(instance)
            .accessToken(accessToken)
            .useStreamingApi()
            .build()

        client.streaming.federatedPublic(
            accessToken = accessToken,
            onlyMedia = false,
            callback = ::println
        ).use {
            Thread.sleep(15_000L)
        }
    }
}
